server/embed/config_logging.go

```diff
@@ -223,22 +183,32 @@ func (cfg *Config) setupLogging() error {
 }
 
 // NewZapCoreLoggerBuilder generates a zap core logger builder.
-func NewZapCoreLoggerBuilder(lg *zap.Logger, cr zapcore.Core, syncer zapcore.WriteSyncer) func(*Config) error {
+func NewZapCoreLoggerBuilder(lg *zap.Logger) func(*Config) error {
 	return func(cfg *Config) error {
 		cfg.loggerMu.Lock()
 		defer cfg.loggerMu.Unlock()
 		cfg.logger = lg
 		cfg.loggerConfig = nil
-		cfg.loggerCore = cr
-		cfg.loggerWriteSyncer = syncer
-
-		grpcLogOnce.Do(func() {
-			if cr != nil && syncer != nil {
-				grpclog.SetLoggerV2(logutil.NewGRPCLoggerV2FromZapCore(cr, syncer))
-			} else {
-				grpc_zap.ReplaceGrpcLoggerV2(cfg.logger.Named("grpc"))
-			}
-		})
+		cfg.loggerCore = nil
+		cfg.loggerWriteSyncer = nil
 		return nil
 	}
 }
+
+// SetupGlobalLoggers configures 'global' loggers (grpc, zapGlobal) based on the cfg.
+//
+// The method is not executed by embed server by default (since 3.5) to
+// enable setups where grpc/zap.Global logging is configured independently
+// or spans separate lifecycle (like in tests).
+func (cfg *Config) SetupGlobalLoggers() {
```

Comment:
A small future refactor suggestion: Since this is no longer called by embed by default should we move it to be closer to the `etcdmain` package instead? Or is there maybe a reason we want to keep it here?

Reply:
All direct users of Embed servers might want to call that API in similar way to how etcd does. 
I decided to change existing behavior, but I think we should make the costs of getting previous behavior as low as possible. 
To some degree Embed server is (the only) part of public API of `go.etcd.io/etcd/server` module.